From d0a9112753248b67e8ad8c09e554866f1a74119c Mon Sep 17 00:00:00 2001
From: Felix Radensky <fradensky@nvidia.com>
Date: Mon, 16 Oct 2023 17:21:03 +0000
Subject: [PATCH] gpio: mlxbf3: Fix error message

Fix the following error message produced by gpio-mlxbf3 driver:

mlxbf3_gpio MLNXBF33:01: IRQ index 0 not found

Bluefield-3 SOC has 2 GPIO controllers, but only one of them
is configured in ACPI table to support GPIO interrupts. By using
platform_get_irq_optional() the error message regarding missing
IRQ support is avoided.

Signed-off-by: Felix Radensky <fradensky@nvidia.com>
Reviewed-by: Vadim Pasternak <vadimp@nvidia.com>
---
 drivers/gpio/gpio-mlxbf3.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/gpio/gpio-mlxbf3.c b/drivers/gpio/gpio-mlxbf3.c
index 0a5f241a8..5fe57ae02 100644
--- a/drivers/gpio/gpio-mlxbf3.c
+++ b/drivers/gpio/gpio-mlxbf3.c
@@ -221,7 +221,7 @@ static int mlxbf3_gpio_probe(struct platform_device *pdev)
 	gc->owner = THIS_MODULE;
 	gc->add_pin_ranges = mlxbf3_gpio_add_pin_ranges;
 
-	irq = platform_get_irq(pdev, 0);
+	irq = platform_get_irq_optional(pdev, 0);
 	if (irq >= 0) {
 		girq = &gs->gc.irq;
 		gpio_irq_chip_set_chip(girq, &gpio_mlxbf3_irqchip);
-- 
2.14.1

